### DISCUSSION TITLE
GNN when only edge attributes are important [V2]

### DISCUSSION THREAD

**Discussion by MehdiZouitine** [pyg-team/pytorch_geometric#4976] [Q&A]
Hello,

I've been stuck with this problem for a few days.
I have a graph where only the weights of the edges are important (TSP with the cost of the edges between each city).
Is there a GNN layer that can learn a good node representation from the edges only?
<img width="434" alt="image" src="https://user-images.githubusercontent.com/39647484/178578733-e8812aa7-d5d0-4449-a3af-7edef0e7f58e.png">

Another simple problem that I can't solve: 
Knowing a graph with weighted edges : train a GNN to predict the total weight of the graph.
Of course I can do the sum and not use GNN because the problem is trivial.
I would just like to know if a GNN is able to handle a graph with only edge features.

I have tried several types of graph convolution but it doesn't work...

Thanks you very much

**Comment by wsad1** [comment]
One way to handle this would be to create some `node` features 
1. One feature could be the sum of edge features/weights incident on a node. A similar approach is taken in the [ogb-protiens](https://ogb.stanford.edu/docs/nodeprop/#ogbn-proteins) which doesn't have node features.
2. Add node statistics like degree, centrality as features. Read more in 2.1.1 of the graph representation learning [book](https://www.cs.mcgill.ca/~wlh/grl_book/files/GRL_Book.pdf)
